src/Symfony/Bundle/WebProfilerBundle/CSP/ContentSecurityPolicyHandler.php

```diff
@@ -0,0 +1,219 @@
+            return [
+                'csp_script_nonce' => $request->headers->get('X-WebProfiler-Script-Nonce'),
+                'csp_style_nonce' => $request->headers->get('X-WebProfiler-Style-Nonce'),
+            ];
+        }
+
+        if ($response->headers->has('X-WebProfiler-Script-Nonce') && $response->headers->has('X-WebProfiler-Style-Nonce')) {
+            return [
+                'csp_script_nonce' => $response->headers->get('X-WebProfiler-Script-Nonce'),
+                'csp_style_nonce' => $response->headers->get('X-WebProfiler-Style-Nonce'),
+            ];
+        }
+
+        $nonces = [
+            'csp_script_nonce' => $this->generateNonce(),
+            'csp_style_nonce' => $this->generateNonce(),
+        ];
+
+        $response->headers->set('X-WebProfiler-Script-Nonce', $nonces['csp_script_nonce']);
+        $response->headers->set('X-WebProfiler-Style-Nonce', $nonces['csp_style_nonce']);
+
+        return $nonces;
+    }
+
+    /**
+     * Cleanup temporary headers and updates Content-Security-Policy headers.
+     *
+     * This method should be called on KernelEvents::RESPONSE
+     *
+     * @param Request  $request
+     * @param Response $response
+     *
+     * @return array Nonce used by the bundle in Content-Security-Policy header
+     */
+    public function onKernelResponse(Request $request, Response $response)
+    {
+        $nonces = $this->getNonces($request, $response);
+        $this->cleanHeaders($response);
+        $this->updateCSPheaders($response, $nonces);
+
+        return $nonces;
+    }
+
+    /**
+     * @param Response $response
+     */
+    private function cleanHeaders(Response $response)
+    {
+        $response->headers->remove('X-WebProfiler-Script-Nonce');
+        $response->headers->remove('X-WebProfiler-Style-Nonce');
+    }
+
+    /**
+     * Updates Content-Security-Policy headers in a response.
+     *
+     * @param Response $response
+     * @param array    $nonces
+     *
+     * @return array
+     */
+    private function updateCSPheaders(Response $response, array $nonces = array()) {
+        $nonces = array_replace([
+            'csp_script_nonce' => $this->generateNonce(),
+            'csp_style_nonce' => $this->generateNonce(),
+        ], $nonces);
+
+        $scriptRule = false;
+        $styleRule = false;
+
+        $headers = $this->getCSPHeaders($response);
+
+        foreach($headers as $header => $directives) {
+            if ($this->CSPpreventsUnsafeInline($directives, 'script-src')) {
+                $scriptRule = true;
+                $headers[$header]['script-src'][] = sprintf('\'nonce-%s\'', $nonces['csp_script_nonce']);
+            }
+            if ($this->CSPpreventsUnsafeInline($directives, 'style-src')) {
+                $styleRule = true;
+                $headers[$header]['style-src'][] = sprintf('\'nonce-%s\'', $nonces['csp_style_nonce']);
+            }
+        }
+
+        if (!$styleRule && !$scriptRule) {
```

Comment:
Looking at the code of this method, I think you could replace these two flags (`$styleRule` and `$scriptRule`) by a single flag.